custom/icds_reports/static/js/angular-services/navigation.service.js

```diff
@@ -0,0 +1,20 @@
+window.angular.module('icdsApp').factory('navigationService', function () {
```

Comment:
This is a common pattern I've been employing throughout this project. Namely, finding some non-trivial chunk of code that appears verbatim in more than one place in dashboard front end code and then porting it into a service.

In many cases I haven't actually ported all usages yet, but plan to as I touch the pages that use them so I can ensure everything is tested.